src/DotNetCore.CAP.RabbitMQ/CAP.RabbiMQOptions.cs

```diff
@@ -84,6 +85,11 @@ public class RabbitMQOptions
         /// </summary>
         public Func<BasicDeliverEventArgs, List<KeyValuePair<string, string>>>? CustomHeaders { get; set; }
 
+        /// <summary>
+        /// If you need to get additional native delivery args, you can use this function to write into <see cref="CapHeader"/>.
+        /// </summary>
+        public Func<BasicDeliverEventArgs, IServiceProvider, List<KeyValuePair<string, string>>>? CustomHeadersBuilder { get; set; }
```

Comment:
I suggest Add `[Obsolete("xxxxx")]` for old `CustomHeaders` option, we will  remove old option at the feature